backend/common/src/main/java/ai/verta/modeldb/common/futures/FutureJdbi.java

```diff
@@ -84,4 +84,61 @@ private <T extends Exception> InternalFuture<Void> useHandleOrTransaction(
 
     return InternalFuture.from(promise);
   }
+
+  public <R, T extends Exception> Future<R> handleCompose(HandleCallback<Future<R>, T> callback) {
```

Comment:
Should we mark the `InternalFuture` returning methods as deprecated?

Reply:
I think we want to wait and let the new one bake a little bit before we commit all the way on this. Should be able to do this as soon as we're happy with how the new one works and feels.